csrc/quantization/cutlass_w8a8/broadcast_load_epilogue_c2x.hpp

```diff
@@ -283,7 +311,26 @@ struct VisitorColOrScalarBroadcast {
         pred(i) = get<0>(tC_cCol(i)) < m;
       }
 
-      if (params_ptr->col_broadcast) {
+      auto fill_dst = [&](Element const& value) {
+        auto dst_v = filter(tC_rCol);
+
+        CUTLASS_PRAGMA_UNROLL
+        for (int i = 0; i < size(dst_v); ++i) {
+          if (pred(i)) {
+            dst_v(i) = value;
+          }
+        }
+      };
+
+      if constexpr(EnableNullptr) {
+        if (params_ptr->ptr_col == nullptr) {
+          fill_dst(params_ptr->null_default);
+          return;
+        }
+      }
+
+      // if nullptr is enabled but not present, always col broadcast
+      if (EnableNullptr || params_ptr->col_broadcast) {
         // In this case we are loading from a column vector and broadcasting
         copy_if(pred, tC_gCol, tC_rCol);
       } else {
```

Comment:
you created a `fill_dst` utility thats a copy of this `else` statement, why not use it there too?

Reply:
Tyler told me to make as few changes as possible to existing codepaths. We're also hoping to revert this soon once my CUTLASS PR lands